tasmota/xsns_53_sml.ino

```diff
@@ -1553,6 +1553,9 @@ void SML_Decode(uint8_t index) {
             // calc difference
             dtimes[dindex] = millis();
             double vdiff = meter_vars[ind - 1] - dvalues[dindex];
+            if(dvalues[dindex] > 0) {
+              dvalid[vindex] = 1;
```

Comment:
I'm not sure but just wondering why testing 'dvalue' just before modifying it.
There might be a good reason of course but my instinctive reaction would have been to test it after the next assignment

Reply:
my intention was to only set dvalid to true when the actual derivative has been calculated

Given the initialization (which was lacking), the dvalue(s) == 0 means there was NO calculation perofmed yet.

After first calculation dvalues will always be > 0 because it will contain the value of the meter.


alternatively one could introdce a flag variable and track the pristiness of dvalues separately; it's just optimisation to use dvalues also as a flag